pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -443,6 +444,7 @@ func (configgen *ConfigGeneratorImpl) createGatewayHTTPFilterChainOpts(node *mod
 				},
 				ServerName:          EnvoyServerName,
 				HttpProtocolOptions: httpProtoOpts,
+				StripPortMode:       &hcm.HttpConnectionManager_StripAnyHostPort{StripAnyHostPort: features.StripHostPort},
```

Comment:
nit: can we leave this unset if the flag is not set? I think this would allow people to override with envoyfilter, where-as with the way it is now we cannot.

Reply:
Sure